src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -1031,7 +1100,14 @@ public void ScheduleCommit()
                         if (_index._cancellationTokenSource.IsCancellationRequested)
                         {
                             //perform the commit
-                            _index._writer?.Commit();
+                            if (_index._writer is ExamineIndexWriter examineIndexWriter)
```

Comment:
This same check is performed a lot, would be better to encapsulate this logic so there's no duplication.